lbry/lbry/extras/cli.py

```diff
@@ -276,14 +249,69 @@ def __exit():
         loop.run_until_complete(loop.shutdown_asyncgens())
 
 
+def setup_logging(args: list, conf: Config):
+    default_formatter = logging.Formatter("%(asctime)s %(levelname)-8s %(name)s:%(lineno)d: %(message)s")
+    file_handler = logging.handlers.RotatingFileHandler(
+        conf.log_file_path, maxBytes=2097152, backupCount=5
+    )
+    file_handler.setFormatter(default_formatter)
+    log.addHandler(file_handler)
+    logging.getLogger('torba').addHandler(file_handler)
+
+    if 'quiet' in args and not args.quiet:
+        handler = logging.StreamHandler()
+        handler.setFormatter(default_formatter)
+        log.addHandler(handler)
+        logging.getLogger('torba').addHandler(handler)
+        logging.getLogger('torba').setLevel(logging.INFO)
+
+    logging.getLogger('aioupnp').setLevel(logging.WARNING)
+    logging.getLogger('aiohttp').setLevel(logging.CRITICAL)
+
+    if conf.share_usage_data:
+        loggly_handler = get_loggly_handler()
+        loggly_handler.setLevel(logging.ERROR)
+        log.addHandler(loggly_handler)
+
+
+def prepare_directory(dir_path: str):
+    if not os.path.isdir(dir_path):
+        pathlib.Path(dir_path).mkdir(parents=True, exist_ok=True)
+    testfile = tempfile.TemporaryFile(dir=dir_path)
+    testfile.close()
+
+
+def change_to_default_if_inaccessible(parsed_path: str, path_obj: Path, conf: Config):
+    assert path_obj.metavar in ('DIR', 'FILE'), "Path must be either to a directory or a file"
+    handled_errors = (OSError, IOError, PermissionError)
+    is_file = path_obj.metavar == 'FILE'
+    default_path = path_obj.default
+    dir_path = os.path.basename(parsed_path) if is_file else parsed_path
+    default_dir_path = os.path.basename(default_path) if is_file else default_path
+    try:
+        prepare_directory(dir_path)
```

Comment:
if `prepare_directory` returns the directory path when it doesn't fail, you can just `return prepare_directory(dir_path)` here, and not need the nested try/except (only logging the warning in the first except, and the critical in the second non-nested except before re-raising)